core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1511,9 +1518,78 @@ class KafkaApis(val requestChannel: RequestChannel,
   }
 
   def handleTxnOffsetCommitRequest(request: RequestChannel.Request): Unit = {
-    val emptyResponse = new java.util.HashMap[TopicPartition, Errors]()
-    def createResponse(throttleTimeMs: Int): AbstractResponse = new TxnOffsetCommitResponse(throttleTimeMs, emptyResponse)
-    sendResponseMaybeThrottle(request, createResponse)
+    val header = request.header
+    val txnOffsetCommitRequest = request.body[TxnOffsetCommitRequest]
+    // reject the request if not authorized to the group
+    if (!authorize(request.session, Read, new Resource(Group, txnOffsetCommitRequest.consumerGroupId))) {
```

Comment:
Discussed offline. It might be tough to consolidate all of the logic with the normal offset commit handler, but perhaps we can extract some common functions such as the authorization and topic existence checks.